### PR TITLE
OSDOCS-3263: Update Deleting ROSA STS cluster documentation to clarify when to delete roles

### PR DESCRIPTION
**Pull request by bmcelvee**
Version(s):
4.12+ 

Issue:
https://issues.redhat.com/browse/OSDOCS-3263

Link to docs preview:
https://54543--docspreview.netlify.app/openshift-rosa/latest/rosa_install_access_delete_clusters/rosa-sts-deleting-cluster.html#rosa-deleting-cluster_rosa-sts-deleting-cluster

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

**Review by arendej — APPROVED** [review]
Looks good, thanks!

**Comment by bergerhoffer** [comment]
The `branch/enterprise-4.13` label has been added to this PR.

This is because your PR targets the `main` branch and is labeled for enterprise-4.12. And any PR going into `main` must also target the latest version branch (enterprise-4.13).

If the update in your PR does NOT apply to version 4.13 onward, please re-target this PR to go directly into the appropriate version branch or branches (enterprise-4.x) instead of main.

**Comment by bmcelvee** [comment]
Hi @yuwang-RH! Would you mind giving this PR a review, please? Thanks!

**Comment by yuwang-RH** [comment]
LGTM~

File: modules/rosa-deleting-cluster.adoc [hunk 1]
```diff
@@ -125,8 +125,13 @@ $ rosa delete oidc-provider -c <cluster_id> --mode auto <1>
...
 You can use the `-y` option to automatically answer yes to the prompts.
 ====
-
-. Delete the cluster-specific Operator IAM roles:
++
+. Optional. Delete the cluster-specific Operator IAM roles:
++
+[IMPORTANT]
+====
+The account-wide IAM roles can be used by other ROSA clusters in the same AWS account. You must only remove the roles if they are not required by other clusters.
```
Comment (mburke5678) [inline modules/rosa-deleting-cluster.adoc:133]: s/You must only/Only   Maybe? It seems odd to have a _You must_ in an optional step. I could be wrong.

Comment (mburke5678) [reply modules/rosa-deleting-cluster.adoc:133]: If I really wanted to be picky, I would say _Rnly remove the roles only if they are not required by other clusters._

Comment (bmcelvee) [reply modules/rosa-deleting-cluster.adoc:133]: I like it! Thanks! :)

**Pull request merged**